### PR TITLE
Flower Baseline: FedProx MNIST

### PR DESCRIPTION
**Pull request by charlesbvll**
#### What does this implement/fix? Explain your changes.

Adds a new baseline for FedProx+MNIST, it is a Flower implementation of this [paper](https://arxiv.org/abs/1812.06127).

### CODE REVIEW

File: doc/source/changelog.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ We would like to give our special thanks to all the contributors who made the ne
 
   A new code example (`quickstart_fastai`) demonstrates federated learning with [fastai](https://www.fast.ai/) and Flower. You can find it here: [quickstart_fastai](https://github.com/adap/flower/tree/main/examples/quickstart_fastai).
 
+- **Introduce new Flower Baseline: FedProx MNIST** ([#1513](https://github.com/adap/flower/pull/1513))
```
Comment (danieljanes) [inline doc/source/changelog.md:29]: Move this up to a new "Unreleased" section

File: baselines/flwr_baselines/publications/fedprox_mnist/tests.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Used to test the model and the data partitionning."""
```
Comment (danieljanes) [inline baselines/flwr_baselines/publications/fedprox_mnist/tests.py:1]: The entire module `tests.py` should be renamed to belong to one of the other modules, for example, `model_test.py`. If functions from multiple modules are being tested (e.g., `model` and `dataset`), then we should split this up into multiple test modules, each belonging to one of the existing non-test modules

File: baselines/flwr_baselines/publications/fedprox_mnist/model.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.parameter import Parameter
+from torch.utils.data import DataLoader
+
+
+class Net(nn.Module):
+    """Convolutional Neural Network architecture as described in McMahan 2017
+    paper :
```
Comment (danieljanes) [inline baselines/flwr_baselines/publications/fedprox_mnist/model.py:15]: Split this onto two separate lines (with an empty line in between - the first sentence in a docstring should not overflow to the next line).

**Review by danieljanes — CHANGES_REQUESTED** [review]
Looks good overall, I added some small comments

**Review by danieljanes — APPROVED** [review]

**Comment by jafermarq** [comment]
This baseline needs to be revisited:
* It should use 1000 clients (instead of 100)
* Different hyperparameters compared to paper? (see Appendix C2: https://arxiv.org/pdf/1812.06127.pdf) . It should use 10 clients per round (instead of full participation. currently is `fraction_fit`=1.0); Default learning rate should be changed to 0.3
